src/main/java/com/minecolonies/coremod/client/gui/WindowResourceList.java

```diff
@@ -54,15 +71,45 @@
     @NotNull
     private final List<BuildingBuilderResource> resources = new ArrayList<>();
 
+    /**
+     * The position of the warehouse clicked on.
+     */
+    @Nullable
+    private final BlockPos warehousePos;
+
+    /**
+     * The compound data to store the warehouse snapshot to.
+     */
+    @Nullable
+    private final CompoundNBT compound;
```

Comment:
Why do we need this field?

Reply:
The snapshot is calculated inside of this class, so we have to pass along the compound in order to save the snapshot.

Reply:
yeah, we pass it along, but do we need it as a field. couldn't see where we needed the field where we didn't already pass it as a param

Reply:
I don't think so as this saving is not directly called through the constructor.
It first goes through `onOpened` which then proceeds to call `pullResourcesFromWarehouse`, that is where the saving happens.

Reply:
I double checked the code, the compound is only used in saving, but do we need that data there at all? Shouldn't we write completely fresh data anyway?

Reply:
technically, we could give the message only the warehouse snapshot data, colony id and building id (which are needed extra anyway to identify)